webserver/static/scripts/profile.js

```diff
@@ -0,0 +1,33 @@
+const $ = require("jquery");
+
+$(document).ready(function () {
+    let apiKey = $("#api-key");
+    if (apiKey) {  // Only when key is displayed (user is viewing their own page)
+
+        // Not showing confirmation dialog if there's no active key (key value is not displayed).
+        let ignoreConfirmation = apiKey.css("display") === "none";
+
+        $("#btn-generate-api-key").click(function () {
+            if (ignoreConfirmation || confirm('Are you sure you want to generate new API key?\n' +
```

Comment:
I really don't like javascript popups. Can we do it with new pages or javascript to change the contents of the page?